life_expectancy/cleaning.py

```diff
@@ -0,0 +1,188 @@
+import re
+import os
+import argparse
+from typing import TypeVar, Optional
+import pandas as pd
+
+Choosable = TypeVar("Choosable", str, int, float, bool, object)
+COMBINED_COLUMN = 0  # Index of the combined column
+
+
+class RegionNotFoundError(ValueError):
+    """
+    Raised when a specified region is not found in the dataset.
+    """
+    def __init__(self, region: str, valid_regions: list):
+        self.region = region
+        self.valid_regions = valid_regions
+        self.message = (
+            f"Error: The region '{self.region}' does not exist in the data. "
+            f"Valid regions include: {', '.join(self.valid_regions[:5])}."
+        )
+        super().__init__(self.message)
+
+
+def get_base_directory() -> str:
+    """
+    Determines the base directory where the script is located.
+    """
+    return os.path.dirname(os.path.abspath(__file__))
+
+
+def load_data() -> Optional[pd.DataFrame]:
+    """Loads a text file and raises exceptions if case something goes wrong."""
+    data_file_path = os.path.join(get_base_directory(), 'data', 'eu_life_expectancy_raw.tsv')
+    
+    try:
+        data = pd.read_csv(data_file_path, sep='\t')
+        print("TSV file loaded successfully!")
+        return data
+    except FileNotFoundError:
+        print(f"Error: The file '{data_file_path}' was not found.")
+    except pd.errors.EmptyDataError:
+        print("Error: The file is empty.")
+    except pd.errors.ParserError:
+        print("Error: There was a parsing error while reading the file.")
+    
+    return None
+
+
+def split_combined_column(data: pd.DataFrame) -> pd.DataFrame:
+    """Splits the combined column into four separate columns: 'unit', 'sex', 'age', and 'region'."""
+    data_split_columns = data.iloc[:, COMBINED_COLUMN].str.split(',', expand=True)
+    data_split_columns.columns = ['unit', 'sex', 'age', 'region']
+   
+    # Drop the combined column from the original dataframe
+    data_reduced = data.drop(columns=[data.columns[0]])   
+    splitted_data = pd.concat([data_split_columns, data_reduced], axis=1)
+    
+    return splitted_data
+
+    
+def unpivot_dataframe(data: pd.DataFrame) -> pd.DataFrame:
+    """Unpivots the DataFrame to have 'year' and 'value' columns."""
+    unpivoted_data = data.melt(id_vars=['unit', 'sex', 'age', 'region'],
+                               var_name='year',
+                               value_name='value')
+    return unpivoted_data
+
+
+def transform_data(data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Transforms the original DataFrame by splitting the combined column into four 
+    columns and unpivoting the DataFrame.
+    """
+    splitted_data = split_combined_column(data)
+    new_data = unpivot_dataframe(splitted_data)
+    
+    return new_data
+
+
+def remove_column_spaces(data: pd.DataFrame, column_name: str) -> pd.Series:
+    """Removes the spaces that might exist in the column's values."""
+    return data[column_name].str.strip()
```

Comment:
Nit: Not sure if you need to wrap that in a function, yet mostly a matter of taste and code structure